content/blog/sivayanama-simplifying-open-api-specification-with-reusable-common-data-models.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Simplifying OpenAPI specification with reusable common data models
+date: 2023-08-26T08:06:19.141Z
+author: BalaSubramanian Vetrivel
+authorimage: /img/vetrivel-balasubramanian1-photo.jpg
+disable: false
+tags:
+  - API, OpenAPI, OpenAPI specification, OpsRamp API, Yaml
+---
+The [OpenAPI specification](https://www.openapis.org)  is one of the most widely followed API contracts. It is language-agonistic.
+With the help of these OpenAPI specifications, clients can understand the API and invoke them without having access to the code or worrying about the implementation details. 
+
+At times, this open specification file is becoming too complex to manage and understand. In this article, I will discuss the techniques to simplify the open specification with loosely coupled reusable data models.
+
+## Items of OpenAPI specification
+OpenAPI specifications files have many definitions in them. However, the below list of entities are typically bigger in terms of the number of lines and tends to be reused in the specification file.
+
+- schemas
+- pathitems
+- parameters
+- requestBodies
+- responses
+- headers
+- examples
+- linkscallbacks
+
+I will discuss four approaches to deal with these definitions.
+
+## In line definition
+The definition is inline right at the reference point as shown below.
+
+```yaml
+schema:
+  type: object
+  properties:
+    id:
+      type: string
+    name:
+      type: string
+```
+
+## In line inside components object
+Components object in the OpenAPI specification is the home of reusable object definitions. However, these defined objects must be explicitly referred to outside of the components section wherever required.
+
+```yaml
+components:
+  parameters:
+    tenantId:
+      in: path
+      name: tenantId
+      schema:
+        type: string
+      required: true
+      description: Describes the clientId or mspId of tenant
+```
+
+*$ref* is one of the fixed fields in the schema. It is a string value which refers to other components in the OpenAPI document, internally and externally. The above defined *tenantId* parameter can be referred to below: 
```

Comment:
Change "which" to "that"

(information on using "which" versus "that": https://www.grammarly.com/blog/which-vs-that/ )